Sources/Regex/CustomCharClass.swift

```diff
@@ -1,14 +1,18 @@
 
-// TODO: struct? source ranges? ASTValue? etc?
-public enum CustomCharacterClass: Hashable {
-  // TODO: store the start too, relevant for e.g. inversion
-  indirect case setOperation([Member], SetOp, [Member])
-  indirect case set([Member])
+// TODO: source ranges? ASTValue? etc?
+public struct CustomCharacterClass: Hashable {
+  public var start: Start
+  public var members: [Member]
 
   public enum Member: Hashable {
+    /// A nested custom character class `[[ab][cd]]`
     case custom(CustomCharacterClass)
+    /// A character range `a-z`
     case range(Atom, Atom)
+    /// A single character or escape
     case atom(Atom)
+    /// A binary operator applied to sets of members `abc&&def`
+    case setOperation([Member], SetOp, [Member])
```

Comment:
This all looks much nicer! It's also easy to extend with source ranges, I like it!